### PR TITLE
[Feat]: Sidechain split

### PR DESCRIPTION
**Pull request by phijfry**

### CODE REVIEW

File: contracts/sidechain/L1Coordinator.sol [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        address _balToken,
+        address _auraToken,
+        address _auraOFT
+    ) NonblockingLzApp(_lzEndpoint) {
+        booster = _booster;
+        balToken = _balToken;
+        auraToken = _auraToken;
+        auraOFT = _auraOFT;
+
+        IERC20(_balToken).approve(_booster, type(uint256).max);
```
Comment (0xahtle7) [inline contracts/sidechain/L1Coordinator.sol:61]: safeApprove

File: contracts/sidechain/AuraProxyOFT.sol [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    ------------------------------------------------------------------- */
+
+    constructor(
+        address _lzEndpoint,
+        address _token,
+        address _locker
+    ) ProxyOFT(_lzEndpoint, _token) {
+        locker = _locker;
+
+        IERC20(_token).approve(_locker, type(uint256).max);
```
Comment (0xahtle7) [inline contracts/sidechain/AuraProxyOFT.sol:33]: safeApprove

File: contracts/sidechain/L2Coordinator.sol [hunk 3]
```diff
@@ -66,7 +73,7 @@ contract Coordinator is OFT, CrossChainConfig {
     function mint(address _to, uint256 _amount) external {
         require(msg.sender == booster, "!booster");
         uint256 amount = (_amount * mintRate) / 1e18;
-        _transfer(address(this), _to, amount);
+        IERC20(auraOFT).transfer(_to, amount);
```
Comment (0xahtle7) [inline contracts/sidechain/L2Coordinator.sol:76]: safeTransfer

**Review by 0xahtle7 — APPROVED** [review]
Small details but LGTM

File: contracts/sidechain/AuraProxyOFT.sol [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        uint16 _srcChainId,
+        bytes memory _srcAddress,
+        uint64 _nonce,
+        bytes memory _payload
+    ) internal virtual override {
+        if (CCM.isCustomMessage(_payload)) {
+            // The payload is a specific cross chain message we decode
+            // the type to determine what the message is an continue
+            CCM.MessageType messageType = CCM.getMessageType(_payload);
+            if (messageType == CCM.MessageType.LOCK) {
```
[the remark this answers is not in the document]
Comment (phijfry) [reply contracts/sidechain/AuraProxyOFT.sol:66]: `FEES` gets processed by `L1Coordinator` and `FEES_CALLBACK` gets processed by `L2Coordinator`

**Review by Hephyrius — APPROVED** [review]

**Pull request merged**